qcelemental/dft_functional_info.py

```diff
@@ -0,0 +1,59 @@
+"""
+Contains metadata about density functionals
+"""
+
+from typing import Dict
+
+from pydantic import Field
+
+from .models import ProtoModel
+
+
+class DFTFunctional(ProtoModel):
+    ansatz: int = Field(
+        ...,
+        description="Maximum functional derivative with respect to the density. This is 2 for meta-GGAs, 1 for GGAs, and 0 otherwise.)",
+    )
+    deriv: int = Field(..., description="Maximum derivative to compute.")
+    c_hybrid: bool = Field(..., description="Requires wavefunction correlation?")
+    x_hybrid: bool = Field(..., description="Requires exact exchange?")
+    c_lrc: bool = Field(..., description="Contains range-separated correlation?")
+    x_lrc: bool = Field(..., description="Contains range-separated exchange?")
+    nlc: bool = Field(..., description="Does this functional need non-local correlation?")
+
+
+class DFTFunctionalInfo:
+    """Information about DFT functionals.
+
+    Parameters
+    ----------
+    context : {'PSI4'}
+        Origin of loaded data.
+
+    Attributes
+    ----------
+    functionals : Dict[str, DftFunctional]
+        dictionary of (lower-case) functional name to info
+
+    name : str
+        The name of the context ('PSI4')
+    """
+
+    def __init__(self, context: str = "PSI4"):
+        if context != "PSI4":
+            raise KeyError(f"Context set as '{context}'," + " only contexts {'PSI4'} are currently supported")
```

Comment:
Ah, I thought libxc had already been considered and rejected to avoid assigning blame to the source (xc) for aftermarket additions (p4). p4 drawing from xc 4.3.4, not that you can access that value programmatically.

I've no strong opinions. Would pylibxc help with direct access?

Reply:
I guess my main point is that there isn't really a "context" for this since these are universal definitions (or as close to it as you can get). The data here is the ground truth and we avoid quibbles over items like VWN by not providing that level of detail. You could argue that this will evolve and may end up getting into those quibbles.

The real name for this is LibXC + some extra Psi4 stuff, and it is good to take the union. Going through Psi4 because its very easy to obtain a compiled version of the data.

This is arguing a bit that this data might go more aptly into a package. "CMS-ML Data" or similar.

Reply:
Following up on in-person discussion, the current plan is to merge this PR as-is, then relegate obscure data to an obscure subfolder at a later date.